packages/fiber/src/core/utils.ts

```diff
@@ -23,6 +23,17 @@ export function calculateDpr(dpr: Dpr) {
   return Array.isArray(dpr) ? Math.min(Math.max(dpr[0], window.devicePixelRatio), dpr[1]) : dpr
 }
 
+export const filterKeys = (obj: any, prune = false, ...keys: string[]) => {
+  // @ts-ignore
+  const keysToSelect = new Set(keys.flat())
+  // @ts-ignore
+  return Object.fromEntries(Object.entries(obj).filter(([key]) => keysToSelect.has(key) === !prune))
+}
```

Comment:
TS doesn't like my usage of `keys.flat()` and `Object.fromEntries` because they're relatively new web APIs.

Reply:
This probably should be using Generics to type, similar to how you can write a TS omit or pick function.

Reply:
Not sure how I would type the result, as it can either prune or pick.